src/resource_clients/request_queue.ts

```diff
@@ -79,7 +88,9 @@ export class RequestQueueClient extends ResourceClient {
         const response = await this.httpClient.call({
             url: this._url('head'),
             method: 'GET',
-            timeout: this.timeoutMillis,
+            timeout: this.timeoutMillis
+                ? Math.min(SMALL_TIMEOUT_SECS * 1000, this.timeoutMillis)
```

Comment:
yeah, it's confusing, millis are now seconds, so you'd need to multiply this too :sweat_smile:

Please revisit this :pray: